Libraries/Dijkstra/Dijkstra.cpp

```diff
@@ -38,31 +56,31 @@ void Dijkstra::updateMatrix(const Map tiles_map) {
     // Then checking and updating weights in the matrix_.
     if (tiles_map.getTile(current_coord_->x, current_coord_->y).ableToGoNorth()
     && !tiles_map.getTile(current_coord_->x - 1, current_coord_->y).isBlack()) {
-      if (matrix_[current_coord_->x - 1][current_coord_->y].weight > matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x - 1, current_coord_->y).getWeight()) {
+      if (matrix_[current_coord_->x][current_coord_->y].weight + tiles_map.getTile(current_coord_->x - 1, current_coord_->y).getWeight() < matrix_[current_coord_->x - 1][current_coord_->y].weight) {
```

Comment:
I still believe this could be made more compact, but for this PR is fine. Could you please add a comment at the top of this function like the other TODO you added? For us to chat about this more.

Reply:
Okay, I will.